datalad/interface/publish_handle.py

```diff
@@ -63,10 +63,16 @@ def __call__(self, target, handle=curdir, url=None, remote=None):
         local_handle_repo = get_repo_instance(
             abspath(expandvars(expanduser(handle))), HandleRepo)
 
-        # first try: local path
-        if not isdir(abspath(expandvars(expanduser(target)))):
+        if target is None:
+            # assume url is a valid repo to push to;
+            assert url is not None
+        # TODO: direct ssh-publishing for single handle
+        elif not isdir(abspath(expandvars(expanduser(target)))):
+            # unexpected target type or invalid path
             raise RuntimeError("Can't handle target %s" % target)
-        target = abspath(expandvars(expanduser(target)))
+        else:
+            # target is local path
+            target = abspath(expandvars(expanduser(target)))
```

Comment:
multiplicitu of `expandvars(expanduser())` begs for defining/using e.g. `expandpath`